Sources/DeploymentTargetAWSLambda/AWSStuff.swift

```diff
@@ -0,0 +1,473 @@
+        try! awsClient.syncShutdown()
+    }
+    
+    
+    
+    
+    /// Creates a new HTTP API Gateway in the specified AWS region.
+    /// - returns: on success, the newly created API's id
+    func createApiGateway(protocolType: SotoApiGatewayV2.ApiGatewayV2.ProtocolType) throws -> String {
+        let apiId = try apiGateway.createApi(ApiGatewayV2.CreateApiRequest(
+            name: "apodini-tmp-api", // doesnt matter will be replaced when importing the openapi spec
+            protocolType: protocolType
+        )).wait().apiId!
+        _ = try apiGateway.createStage(ApiGatewayV2.CreateStageRequest(
+            apiId: apiId,
+            autoDeploy: true,
+            stageName: "$default"
+        )).wait()
+        return apiId
+    }
+    
+    
+    
+    
+    /// - parameter s3BucketName: name of the S3 bucket the function should be uploaded to
+    /// - parameter s3ObjectFolderKey: key (ie path) of the folder into which the function should be uploaded
+    func deployToLambda(
+        deploymentStructure: DeployedSystemStructure,
+        openApiDocument: OpenAPI.Document,
+        lambdaExecutableUrl: URL,
+        lambdaSharedObjectFilesUrl: URL,
+        s3BucketName: String,
+        s3ObjectFolderKey: String,
+        apiGatewayApiId: String
+    ) throws {
+        guard !didRunDeployment else {
+            fatalError("Cannot call '\(#function)' multiple times.")
+        }
+        didRunDeployment = true
+        
+        let accountId = try sts.getCallerIdentity(STS.GetCallerIdentityRequest()).wait().account!
+        
+        logger.notice("Fetching list of all lambda functions in AWS account")
+        let allFunctions = try { [unowned self] () -> [Lambda.FunctionConfiguration] in
+            var retval: [Lambda.FunctionConfiguration] = []
+            var nextMarker: String?
+            repeat {
+                let response = try lambda.listFunctions(Lambda.ListFunctionsRequest(marker: nextMarker)).wait()
+                retval.append(contentsOf: response.functions ?? [])
+                nextMarker = response.nextMarker
+            } while nextMarker != nil
+            return retval
+        }()
+        logger.notice("#functions: \(allFunctions.count) \(allFunctions.map(\.functionArn!))")
+
+        
+        // Delete old functions
+//        do {
+//            let functionsToBeDeleted = allFunctions.filter { $0.functionName!.hasPrefix(Self.lambdaFunctionNamePrefix) }
+//            if !functionsToBeDeleted.isEmpty {
+//                logger.notice("Deleting old apodini lambda functions")
+//                for function in functionsToBeDeleted {
+//                    logger.notice("[SKIPPED] - deleting \(function.functionArn!)")
+//                    //try lambda
+//                    //    .deleteFunction(Lambda.DeleteFunctionRequest(functionName: function.functionName!))
+//                    //    .wait()
+//                }
+//            }
+//        }
```

Comment:
As noted in the last meeting this would be a good idea. In combination with getting the real hostname using using this differentiate between different lambdas of different web services this should work. 
Maybe one more general comment: I would suggest to make the more destructive actions (e.g. removing lambdas) after the deployment of new lambdas was successful. Otherwise we could have an error in the middle of the deployment and might be stuck with a system that is no longer up. I think this might also apply to some of the order the tasks happen here that I might have missed.